node-fs-readfile/cat.js

```diff
@@ -0,0 +1,18 @@
+import { readFile } from 'node:fs/promises';
+
+const fileArray = process.argv.slice(2);
+
+async function print() {
+  await Promise.all(
+    fileArray.map(async (file) => {
+      try {
+        const data = await readFile(`./${file}`, { encoding: 'utf8' });
+        console.log(data);
+      } catch (error) {
+        console.log(error.message);
+      }
+    })
+  );
+}
```

Comment:
I'm not convinced this works. Did you run it multiple times as instructed to make sure it always print in the proper order?

The reason I'm skeptical is because you read and write the file immediately, and Promise.all will run the promises all at the same time. So I would expect the one that finishes first to print first.